src/lib/Persistence/Legacy/Content/Gateway/DoctrineDatabase.php

```diff
@@ -910,6 +910,51 @@ public function loadVersionInfo(int $contentId, ?int $versionNo = null): array
         return $queryBuilder->execute()->fetchAll(FetchMode::ASSOCIATIVE);
     }
 
+    /**
+     * @return array<mixed>
+     */
+    public function loadVersionNoArchivedWithin(int $contentId, int $seconds): array
+    {
+        $cutoffTimestamp = strtotime(sprintf('-%d seconds', $seconds));
+        if (!$cutoffTimestamp) {
```

Comment:
This is not correct. While this edge case would never happen probably, still, if you get timestamp that equals to 0, it's gonna be treated as false.
If you have a function or method that returns nasty union type, you should always use strict comparison, just to make it as a habit.
```suggestion
        if (false === $cutoffTimestamp) {
```